components/header/Header.js

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react';
+import styles from './Header.module.css';
+import logoImg from '@/public/logo.svg';
+import styled from 'styled-components';
+import { FetchUserData } from '@/pages/api/UsersAPI';
+import Link from 'next/link';
+import Image from 'next/image';
+
+const StyledHeader = styled.header`
+  position: ${props => (props.$notFixed ? 'static' : 'fixed')};
+`;
+
+function Header({ notFixed }) {
+  const [userInfo, setUserInfo] = useState();
+  const invisible = { display: 'none' };
+  const visible = { display: 'inherit' };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const userData = await FetchUserData();
+      setUserInfo(userData);
+    };
+    fetchData();
+  }, []);
+
+  if (!userInfo) {
+    return null;
+  }
+
+  return (
+    <StyledHeader $notFixed={notFixed} className={styles.header}>
+      <div className={styles.headerSon}>
+        <Link href="/">
+          <Image src={logoImg} alt="로고" className={styles.logo} />
+        </Link>
+        <a
+          className={styles.loginButton}
+          href="./signin/signin.html"
+          style={userInfo['data'][0]['id'] !== 1 ? visible : invisible}
+        >
+          로그인
+        </a>
+        <div
+          className={styles.userInfo}
+          style={userInfo['data'][0]['id'] === 1 ? visible : invisible}
```

Comment:
style={userInfo['data'][0]['id'] === 1 ? visible : invisible}
          이 코드가 중복돼므로 visibleStyle 이런식으로 볁수로 선언하고 재사용하셔도 좋을것 같아요!

Reply:
잘 이해가 안되는데 혹시 힌트 좀 부탁드려도 될까요 ㅠ??

Reply:
```suggestion
const visibleStyle =  userInfo['data'][0]['id'] === 1 ? visible : invisible
          style={visibleStyle}
```
위처럼 visibleStyle을 변수로 선언하고 재사용하면 좋을것 같습니다:)